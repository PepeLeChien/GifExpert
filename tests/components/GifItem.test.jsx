import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('tests in GifItem', () => {
    
    const title = "Un puÃ±o";
    const url = "https://blablabla.com/saitama.jpg";

    test('should match with snapshot', () => { 
        
        const {container} = render(<GifItem title={title} url={url}/>);

        expect(container).toMatchSnapshot();
    })

    test('should match src and alt in img', () => {

        render(<GifItem title={ title } url={ url } />)
        
        const { src, alt } = screen.getByRole('img');
        
        expect(src).toBe(url);
        expect(alt).toBe(title);
    })

    test('should exist the title in GifItem', () => {

        render(<GifItem title={title} url={url} />)

        expect(screen.getByText(title)).toBeTruthy();

    })


})