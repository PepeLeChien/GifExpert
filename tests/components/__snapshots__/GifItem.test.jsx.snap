// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in GifItem should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Un puño"
      src="https://blablabla.com/saitama.jpg"
    />
    <p>
      Un puño
    </p>
  </div>
</div>
`;
